services/datamanager/builtin/builtin.go

```diff
@@ -512,16 +494,46 @@ func (svc *builtIn) uploadData(cancelCtx context.Context, intervalMins float64)
 			case <-ticker.C:
 				svc.lock.Lock()
 				if svc.syncer != nil {
-					svc.flushCollectors()
-					svc.syncer.SyncDirectory(svc.captureDir)
-					svc.syncAdditionalSyncPaths()
+					svc.sync()
 				}
 				svc.lock.Unlock()
 			}
 		}
 	})
 }
 
+func (svc *builtIn) sync() {
+	svc.flushCollectors()
+	toSync := getAllFilesToSync(svc.captureDir, svc.waitAfterLastModifiedMillis)
+	for _, ap := range svc.additionalSyncPaths {
+		toSync = append(toSync, getAllFilesToSync(ap, svc.waitAfterLastModifiedMillis)...)
+	}
+	for _, p := range toSync {
+		svc.syncer.SyncFile(p)
+	}
+}
+
+// nolint
+func getAllFilesToSync(dir string, lastModifiedMillis int) []string {
+	var filePaths []string
+	_ = filepath.Walk(dir, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return nil
+		}
+		if info.IsDir() {
+			return nil
+		}
+		// If a file was modified within the past waitAfterLastModifiedSecs seconds, do not sync it (data
```

Comment:
[nit] This comment references the name of the field on the builtin service, but that's not what the parameter is called in this function. Can we change either the comment or the parameter name?

Reply:
Done